Add workflow file through CLI

neelasha23 opened the pull request:
Added instructions on adding Github workflow file through CLI.

<!-- readthedocs-preview ploomber-doc start -->
----
📚 Documentation preview 📚: https://ploomber-doc--64.org.readthedocs.build/en/64/

<!-- readthedocs-preview ploomber-doc end -->

edublancas commented on doc/user-guide/github.md:
```diff
@@ -69,4 +76,28 @@ Without `--watch`, the logs will look like this:
+
+Once done, you can monitor progress as discussed [above](monitor). Ensure that the API key is set as Github secret.
+
+In case the [workflow template](https://github.com/edublancas/cloud-template/blob/main/.github/workflows/ploomber-cloud.yaml) has been updated, and you need to re-initialise the application the CLI will prompt your for updating the workflow file:
```
this point is unclear: project re-initialization doesn't happen often. the only case I can think of is if the project gets corrupted.

wouldn't it make more sense to check for an updated workflow file when calling `ploomber-cloud deploy`?

I see two scenarios:

1. if `ploomber-cloud deploy` is running on github actions, then we can juts print a mesage
2. if `ploomber-cloud deploy` is running locally, we can deploy and then prompt the user, asking if they want to update their ploomber-cloud.yaml

neelasha23 replied:
Yes checking during deploy would make more sense. For scenario 1, the message will be displayed in the github actions logs?

edublancas replied:
I thought through this, here's my feedback (note that this workflow is different to what I proposed earlier cause I noticed some potential issues):

- we can add a `ploomber-cloud github` command that can take care of configuring for the first time (if the file doesn't exist) or update the workflow (if it already exists)
- then, when users run `ploomber-cloud init` or `ploomber-cloud deploy`, we can either thell them they can configure github deployment (or update their existing workflow) by running `ploomber-cloud github`

the reason I'm proposing this is because I feel like prompting users for an answer when running `init` or `deploy` might be annoying and unexpected. instead, we can just print some relevant message on `init`/`deploy` and they can decide if they want to configure/update github actions by running `ploomber-cloud github`

Last call is up to you; think what scenarios users will encounter and define the workflow that makes more sense

neelasha23 replied:
Sure! I'll start implementing this and if I find any scenario which would be an issue , I'll discuss and modify accordingly.

edublancas replied:
I think it's ok to tell the user to update only when the contents are different, no need for more complex implementation right now

neelasha23 replied:
Please refer to this [comment](https://github.com/ploomber/cli/pull/22#issuecomment-1904217040)

edublancas commented on doc/user-guide/github.md:
```diff
@@ -69,4 +76,28 @@ Without `--watch`, the logs will look like this:
+
+Once done, you can monitor progress as discussed [above](monitor). Ensure that the API key is set as Github secret.
+
+In case the [workflow template](https://github.com/edublancas/cloud-template/blob/main/.github/workflows/ploomber-cloud.yaml) has been updated, and you need to re-initialise the application the CLI will prompt your for updating the workflow file:
```
let's keep spelling consistent:

re-initialise -> re-initialize
github -> GitHub

neelasha23 replied:
Have added more mentions of `GitHub` in the latest commit.

edublancas commented:
@neelasha23 don't forget to include responses and links to the observations. it's unclear to me if the were addressed or not

neelasha23 commented:
> @neelasha23 don't forget to include responses and links to the observations. it's unclear to me if the were addressed or not

Have addressed all the comments. Most of the code has changed now so I didn't include the links of the previous deleted code.
